service/distros.go

```diff
@@ -15,19 +15,34 @@ import (
 	"github.com/evergreen-ci/evergreen/util"
 	"github.com/evergreen-ci/evergreen/validator"
 	"github.com/evergreen-ci/gimlet"
+	"github.com/evergreen-ci/gimlet/rolemanager"
+	"go.mongodb.org/mongo-driver/bson"
 )
 
 func (uis *UIServer) distrosPage(w http.ResponseWriter, r *http.Request) {
-	distros, err := distro.Find(distro.All)
+	u := MustHaveUser(r)
 
+	distros, err := distro.Find(distro.All.Project(bson.M{"_id": 1}))
 	if err != nil {
-		message := fmt.Sprintf("error fetching distros: %v", err)
+		message := fmt.Sprintf("error fetching distro ids: %v", err)
 		PushFlash(uis.CookieStore, r, w, NewErrorFlash(message))
 		http.Error(w, message, http.StatusInternalServerError)
 		return
 	}
-
 	sort.Sort(&sortableDistro{distros})
+	distroIds := []string{}
+	for _, d := range distros {
+		opts := gimlet.PermissionOpts{Resource: d.Id, ResourceType: evergreen.DistroResourceType}
+		permissions, err := rolemanager.HighestPermissionsForRoles(u.Roles(), evergreen.GetEnvironment().RoleManager(), opts)
```

Comment:
This will do 1 query per distro. Could you instead write a private function that does basically this, but looks up all roles once and checks the highest permission for each distro in a loop?

Reply:
I think I should create a new ticket for this given that the work to make the change is not insignificant and includes changes a gimlet. Can we merge as is for now and just open a ticket to make the gimlet changes?

Reply:
works for me